src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -442,6 +576,64 @@ const FormCreationForm = ({csrftoken, formUuid, formHistoryUrl }) => {
         }
     };
 
+    const onRuleChange = (index, event) => {
+        const { name, value } = event.target;
+        dispatch({
+            type: 'CHANGED_RULE',
+            payload: {name, value, index},
+        });
+    };
+
+    const onRuleDelete = (index) => {
+        dispatch({
+            type: 'DELETED_RULE',
+            payload: {index: index}
+        });
+    };
+
+    const onRuleAdd = () => {
+        dispatch({type: 'ADD_RULE'});
+    };
+
+    const submitRules = async () => {
+        for (let index = 0; index < state.rules.rules.length; index++) {
+            const logicRule = state.rules.rules[index];
+            const isNewRule = !logicRule.uuid;
+            const createOrUpdate = isNewRule ? post : put;
+            const endPoint = isNewRule ? LOGICS_ENDPOINT : `${LOGICS_ENDPOINT}/${logicRule.uuid}`;
+
+            var formLogicResponse = await createOrUpdate(
+                endPoint,
+                csrftoken,
+                logicRule
+            );
+
+            if (!formLogicResponse.ok) {
+                throw new Error('An error occurred while saving the form logic.');
+            }
+
+            var ruleUuid = formLogicResponse.data.uuid;
+            if (isNewRule){
+                dispatch({
+                    type: 'CHANGED_RULE',
+                    payload: {name: 'uuid', value: ruleUuid, index: index}
+                });
+            }
+        }
+
+        if (state.rules.rulesToDelete.length) {
+            for (const ruleUuid of state.rules.rulesToDelete) {
+                // Rules that were added but are not saved in the backend yet don't have a UUID.
+                if (!ruleUuid) return;
+
+                var response = await apiDelete(`${LOGICS_ENDPOINT}/${ruleUuid}`, csrftoken);
+                if (!response.ok) {
+                    throw new Error('An error occurred while deleting logic rules.');
+                }
+            }
```

Comment:
A good candidate for `await Promise.all(...)` to do the HTTP requests in parallel rather than sequentially, improving the UX of end-users.